testsuite/modern-testsuite/src/test/java/org/sonatype/nexus/testsuite/maven/conditional/MavenConditionalRequestIT.java

```diff
@@ -163,7 +147,50 @@ public void conditionalGet() throws Exception {
   }
 
   @Test
+  public void conditionalGetViaGroup() throws Exception {
+    // redirect maven-central
+    {
+      Repository mavenCentral = repositoryManager.get("maven-central");
+      assertThat(mavenCentral, notNullValue());
+      Configuration mavenCentralConfiguration = mavenCentral.getConfiguration();
+      mavenCentralConfiguration.attributes("proxy").set("remoteUrl", "http://localhost:" + upstream.getPort() + "/");
+      repositoryManager.update(mavenCentralConfiguration);
+    }
+
+    // get public group and create client for it
+    Repository mavenPublic = repositoryManager.get("maven-public");
+    assertThat(mavenPublic, notNullValue());
+    Maven2Client publicClient = new Maven2Client(HttpClients.custom().build(), HttpClientContext.create(),
+        resolveUrl(nexusUrl, "/repository/" + mavenPublic.getName() + "/").toURI());
+
+    HttpResponse response;
+
+    response = publicClient.get(RELEASE_ARTIFACT_PATH);
+    EntityUtils.consume(response.getEntity());
+
+    assertThat(response.getStatusLine().getStatusCode(), equalTo(200));
+
+    response = publicClient.getIfNewer(RELEASE_ARTIFACT_PATH, DateUtils.parseDate(LAST_MODIFIED_VALUE));
+    EntityUtils.consume(response.getEntity());
+
+    assertThat(response.getStatusLine().getStatusCode(), equalTo(304));
+
+    response = publicClient.getIfNoneMatch(RELEASE_ARTIFACT_PATH, ETAG_VALUE);
+    EntityUtils.consume(response.getEntity());
+
+    assertThat(response.getStatusLine().getStatusCode(), equalTo(304));
+  }
+
+  @Test
   public void conditionalPut() throws Exception {
+    Repository mavenSnapshots = repositoryManager.get("maven-snapshots");
```

Comment:
This all seems like it should be in another method; having the test method do all this work inline makes it unnecessarily hard to grok what the test is doing. e.g.:
    buildAdminMaven2Client("maven-snapshots")